src/libraries/Common/tests/TestUtilities/TestEventListener.cs

```diff
@@ -116,4 +143,103 @@ protected override void OnEventWritten(EventWrittenEventArgs eventData)
         }
         catch { }
     }
+
+    // From https://gist.github.com/MihaZupan/cc63ee68b4146892f2e5b640ed57bc09
+    private static class ActivityHelpers
```

Comment:
The WebSockets logs I added use activities, because otherwise it is extremely hard to track what was the origins of calls, because everything happens within the same object and most of the times ends up calling the same functions (e.g. ReceiveAsyncPrivate/SendFrameAsync/Dispose etc.). So with ordinary traces you can see that Dispose happened, but unless you also print a stacktrace, you don't know who called it

This can show activities in the "human-readable" format of `/1/1/1/`, e.g.:

```
13:31:51.5332321 /1/5/ - [ReceiveStart] objName: ManagedWebSocket#433292681, memberName: ReceiveAsyncPrivate, bufferLength: 1
13:31:51.5332603 /1/5/ - [WsTrace] objName: ManagedWebSocket#433292681, memberName: ReceiveAsyncPrivate, message: shouldEnterMutex=True, shouldAbortOnCanceled=True
13:31:51.5332784 /1/5/ - [MutexEnter] objName: AsyncMutex#2130326874, memberName: ReceiveAsyncPrivate
13:31:51.5332947 /1/5/ - [WsTrace] objName: ManagedWebSocket#433292681, memberName: ReceiveAsyncPrivate, message: Reading the next frame header
13:31:51.5333431 /1/6/ - [CloseStart] objName: ManagedWebSocket#433292681, memberName: CloseAsyncPrivate
13:31:51.5333610 /1/6/ - [WsTrace] objName: ManagedWebSocket#433292681, memberName: SendCloseFrameAsync, message: SendCloseFrameAsync
13:31:51.5333880 /1/6/1/ -- [SendStart] objName: ManagedWebSocket#433292681, memberName: SendFrameAsync, opcode: Close, bufferLength: 2
13:31:51.5334096 /1/6/1/ -- [MutexEnter] objName: AsyncMutex#102518422, memberName: SendFrameFallbackAsync
13:31:51.5334967 /1/6/1/ -- [MutexExit] objName: AsyncMutex#102518422, memberName: SendFrameFallbackAsync
13:31:51.5335667 /1/6/1/ -- [SendStop] objName: ManagedWebSocket#433292681, memberName: SendFrameFallbackAsync
13:31:51.5336132 /1/6/ - [WsTrace] objName: ManagedWebSocket#433292681, memberName: SendCloseFrameAsync, message: State transition from Open to CloseSent
13:31:51.5336482 /1/6/ - [WsTrace] objName: ManagedWebSocket#433292681, memberName: CloseAsyncPrivate, message: Waiting for a close frame
13:31:51.5336677 /1/6/ - [WsTrace] objName: ManagedWebSocket#433292681, memberName: CloseAsyncPrivate, message: Registering for cancellation
13:31:51.5336841 /1/6/ - [WsTrace] objName: ManagedWebSocket#433292681, memberName: CloseAsyncPrivate, message: Waiting for the _receiveMutex...
13:31:51.5337002 /1/6/ - [MutexContended] objName: AsyncMutex#2130326874, memberName: EnterAsync, queueLength: 1
```

Note: this is just an example of the log format, the actual logging events in the PR might differ